### PR TITLE
chore: Get rid of `LegacyColors` within `SemanticColors`

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR I get rid of the `LegacyColors` within `SemanticColors`.

In addition to that `ConversationListTopBarViewControllerSnapshotTests` I replaced FB's snapshot library with SnapshotTesting library. 

Last but not least I cleaned up `ConversationSystemMessageCell`. Provided a bit of documentation, created a bunch of type alias, and replaced .localized with L10n whenever possible. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by lewandowskit93 — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift [hunk 1]
```diff
@@ -103,6 +97,13 @@ public enum SemanticColors {
 
         // Audio Icon
         static let foregroundAudio = UIColor(light: Asset.Colors.black, dark: Asset.Colors.white)
+
+        // System Message Icon Colors
+        static let foregroundExclamationMarkInSystemMessage = UIColor(light: Asset.Colors.red500Light, dark: Asset.Colors.red500Dark)
+        static let foregroundCheckMarkInSystemMessage = UIColor(light: Asset.Colors.green500Light, dark: Asset.Colors.green500Dark)
+
+        static let backgroundLegalHold = UIColor(light: Asset.Colors.red500Light, dark: Asset.Colors.red500Dark)
+        static let foregroundLegalhold = UIColor(light: Asset.Colors.red500Light, dark: Asset.Colors.red500Dark)
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift]: Do we use `foregroundLegalhold` ?

File: wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift [hunk 2]
```diff
@@ -103,6 +97,13 @@ public enum SemanticColors {
 
         // Audio Icon
         static let foregroundAudio = UIColor(light: Asset.Colors.black, dark: Asset.Colors.white)
+
+        // System Message Icon Colors
+        static let foregroundExclamationMarkInSystemMessage = UIColor(light: Asset.Colors.red500Light, dark: Asset.Colors.red500Dark)
+        static let foregroundCheckMarkInSystemMessage = UIColor(light: Asset.Colors.green500Light, dark: Asset.Colors.green500Dark)
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift:103]: Do we use `foregroundExclamationMarkInSystemMessage` and `foregroundCheckMarkInSystemMessage` ?

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift:103]: Yes!

File: wire-ios/WireCommonComponents/Icons/Autogenerated/WireStyleKit.swift [hunk 3]
```diff
@@ -4384,7 +4385,7 @@ public class WireStyleKit: NSObject {
 
         //// Bezier Drawing
         context.saveGState()
-        context.setAlpha(0.24)
+        context.setAlpha(0.40)
```
Comment (KaterinaWire) [inline wire-ios/WireCommonComponents/Icons/Autogenerated/WireStyleKit.swift:4388]: Why do we need these changes?

File: wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/Helpers/SemanticColors.swift]: I will have to double check on that and if not to remove it. I added those colors as I working on fixing the color of the legal hold icon.

**Review by KaterinaWire — APPROVED** [review]

File: wire-ios/WireCommonComponents/Icons/Autogenerated/WireStyleKit.swift (hunk 3 above)
Comment (agisilaos) [reply wire-ios/WireCommonComponents/Icons/Autogenerated/WireStyleKit.swift:4388]: So we can fix the color/opacity for the color of the icon properly.

**Pull request merged**